sysl2/sysl/seqs/strSet.go

```diff
@@ -0,0 +1,118 @@
+package seqs
+
+import (
+	"sort"
+
+	"github.com/anz-bank/sysl/src/proto"
+)
+
+type (
+	nothing struct{}
+
+	strSet struct {
+		m map[string]nothing
+	}
+)
```

Comment:
Due to the business logic, this strSet will not accept `""`. In this case, I just wrapped `map[string]struct{}` to prevent access the map directly.  Maybe I should name the `strSet` to `patternSet` or some special name.

Reply:
After double check, I will do the special check to avoid empty string in other place.